src/include/gcc.h

```diff
@@ -125,21 +139,37 @@ WT_ATOMIC_CAS_FUNC(size, size_t *vp, size_t old, size_t newv)
 static inline bool
 __wt_atomic_cas_ptr(void *vp, void *old, void *newv)
 {
-    return (WT_ATOMIC_CAS((void **)vp, &old, newv));
+    return (WT_ATOMIC_CAS((void **)vp, &old, newv, WT_ATOMIC_SEQ_CST));
+}
+
+#define WT_ATOMIC_LOAD_PTR(type, vp, memorder) (type *)WT_ATOMIC_LOAD((void **)vp, memorder)
```

Comment:
It is really a pain if we have to do casting every time we read a pointer. Reading a pointer is going to be the predominant pattern we use for C11 in WiredTiger.

Reply:
Why do you need to cast? Doesn't `WT_ATOMIC_LOAD` propagate the type information from `ptr` to the result? The `atomic_load()` function and the `__atomic` builtins are all magically generic and correctly propagate the types of their arguments to their return values, even though that isn't something you can do in a normal C function (but is very common in C++). I'm just proposing that we take advantage of that, rather than throwing away the type-safety it provides.

Reply:
It's a surprise to me that just using `WT_ATOMIC_LOAD` actually works. Thanks for the tip.

Reply:
https://en.cppreference.com/w/c/atomic/atomic_load
> This is a [generic function](https://en.cppreference.com/w/c/language/generic) defined for all [atomic object types](https://en.cppreference.com/w/c/language/atomic) A.

The [generic function](https://en.cppreference.com/w/c/language/generic) link shows how we can do something similar in macros we define, although it requires some trickery to be _as_ generic as those functions without being built into the compiler. I'm not 100% sure it is possible in plain C.